src/realm.h

```diff
@@ -2275,6 +2275,34 @@ RLM_API realm_object_t* realm_dictionary_get_linked_object(realm_dictionary_t*,
  */
 RLM_API bool realm_dictionary_erase(realm_dictionary_t*, realm_value_t key, bool* out_erased);
 
+/**
+ * Return the list of keys stored in the dictionary
+ *
+ * @param out_size number of keys
+ * @param out_keys the list of keys in the dictionary
+ * @return True if no exception occurred.
+ */
+RLM_API bool realm_dictionary_get_keys(realm_dictionary_t*, size_t* out_size, realm_results_t** out_keys);
```

Comment:
Maybe this is conventional in the C API, but returning ownership of newly allocated memory to the caller seems dangerous and IMO at least warrants a note about this in the doc string.

Reply:
It is dangerous, I agree. But it is how the C-API works. The memory has to be freed once it is not used any more. The issue is always the same, we don't have a way to tell the caller if there was a problem due to some exception or if the dictionary was simply empty. We can't use `nullptr` for both. 
I will add a comment in the doc string.